bundles/org.openhab.ui/web/src/pages/settings/things/inbox/inbox-list.vue

```diff
@@ -335,7 +335,8 @@ export default {
         case 'approve':
           progressMessage = 'Approving Inbox Entries...'
           successMessage = `${this.selectedItems.length} entries approved`
-          promises = this.selectedItems.map((i) => this.$oh.api.postPlain('/rest/inbox/' + i + '/approve', i.label))
+          promises = this.selectedItems.map((i) => this.$oh.api.postPlain('/rest/inbox/' + i + '/approve',
```

Comment:
Optional chaining is a pretty recent feature :) It has good support in browsers (Firefox on Android excluded) but not sure if even babel supports it yet. It's also the first occurrence...
Tbh I think I would prefer remaining on the conservative side with something like:
```js
this.selectedItems
  .map((i) => this.inbox.find((e) => e.thingUID === i))
  .map((e) => (e) ? this.$oh.api.postPlain('/rest/inbox/' + e.thingUID + '/approve' : Promise.reject())
```
By the way there's some deeper flaw in this screen (and probably others), when the list is reloaded following a SSE event, `selectedItems` should probably be reset or at least stripped of any items that don't exist anymore, or it will lead to further bugs...

Reply:
Even better:
```js
this.inbox.filter((e) => this.selectedItems.indexOf(e.thingUID) >= 0)
  .map((e) => this.$oh.api.postPlain('/rest/inbox/' + e.thingUID + '/approve', e.label))
```

Reply:
thats good :-)